Controllers/OperationController.cs

```diff
@@ -99,6 +99,9 @@ public class OperationController : ApiController
         [Authorize]
         public async Task<HttpResponseMessage> Invoke(OperationInfo info)
         {
+            // escapeuristring does not escape '#' as it is a valid special character
+            // will not be legit in our case so escaping explicitly.
+            info.Url = Uri.EscapeUriString(info.Url).Replace("#", "%23");
```

Comment:
alright will push this 

// escaping "#" as it may appear in some resource names
            info.Url = info.Url.Replace("#", "%23");

and then it can be tested in staging slot (I tried uploading a cert to repro with the "#" but could not and it just worked for me)